client/src/partials/patient_registration/patient.js

```diff
@@ -131,18 +140,24 @@ angular.module('bhima.controllers')
           return true;
         }
 
-        extractYear = $scope.patient.dob.getFullYear();
+        yearOfBirth = $scope.patient.dob.getFullYear();
       }
 
-      if (extractYear) {
+      if (yearOfBirth) {
+
+	// NOTE: The following checks on the yearOfBirth are never executed with
+	//       the html5+angular date input field since the form value becomes
+	//       undefined when invalid and is caught by the previous check.
+	//       Leaving them in as a precaution in case the input form behavior
+	//       changes in the future.
```

Comment:
This is a really clean message.  Thanks for taking the investment time to produce quality code.